mobile/lib/utils/url_helper.dart

```diff
@@ -90,3 +91,13 @@ String? punycodeDecodeUrl(String? serverUrl) {
 
   return Uri.decodeFull(serverUri.replace(host: decodedHost).toString());
 }
+
+/// Generates the appropriate share URL path for a given shared link.
+///
+/// Returns a path string based on the shared link's slug and key:
+/// - If slug is present: 's/${sharedLink.slug}'
+/// - Otherwise: 'share/${sharedLink.key}'
+String getShareUrlPath(SharedLink sharedLink) {
+  final slug = sharedLink.slug?.trim();
+  return slug?.isNotEmpty == true ? 's/$slug' : 'share/${sharedLink.key}';
+}
```

Comment:
lets move this to the SharedLink class and have a getUrlPath() method that returns a String, similar to toString() that is there currently.